apps/web/src/dialogs/command-palette/command-palette-dialog.tsx

```diff
@@ -0,0 +1,472 @@
+      selectedRef.current?.scrollIntoView({
+        block: "nearest"
+      });
+    }, [selected]);
+
+    function reset() {
+      setSelected(0);
+      setQuery(">");
+      setCommands(getDefaultCommands());
+    }
+
+    const highlighted = db.lookup.fuzzy(
+      prepareQuery(query),
+      commands,
+      "title",
+      {
+        prefix: "<b style='color: var(--accent-foreground)'>",
+        suffix: "</b>"
+      }
+    );
+    const grouped = highlighted.reduce((acc, command, index) => {
+      if (!acc[command.group]) {
+        acc[command.group] = [];
+      }
+      acc[command.group].push({
+        ...command,
+        icon: getCommandIcon(command),
+        index
+      });
+      return acc;
+    }, {} as GroupedCommands);
+
+    return (
+      <Dialog
+        isOpen={true}
+        width={650}
+        onClose={() => {
+          reset();
+          props.onClose(false);
+        }}
+        noScroll
+        sx={{
+          fontFamily: "body"
+        }}
+      >
+        <Flex
+          variant="columnFill"
+          sx={{ mx: 3, overflow: "hidden", height: 350 }}
+          onKeyDown={(e) => {
+            if (e.key == "Enter") {
+              e.preventDefault();
+              const command = commands[selected];
+              if (!command) return;
+              const action = getCommandAction({
+                id: command.id,
+                type: command.type
+              });
+              if (action) {
+                action(command.id);
+                addRecentCommand(command);
+                reset();
+                props.onClose(false);
+              }
+              setSelected(0);
+            }
+            if (e.key === "ArrowDown") {
+              e.preventDefault();
+              setSelected((selected + 1) % commands.length);
+            }
+            if (e.key === "ArrowUp") {
+              e.preventDefault();
+              setSelected((selected - 1 + commands.length) % commands.length);
+            }
+          }}
+        >
+          <>
+            <Field
+              autoFocus
+              placeholder={"Search in notes, notebooks, and tags"}
+              sx={{ mx: 0, my: 2 }}
+              defaultValue={query}
+              onChange={
+                isCommandMode(query) || query.trim().length < 1
+                  ? (e) => {
+                      setSelected(0);
+                      const query = e.target.value;
+                      setQuery(query);
+
+                      const res = search(query);
+                      if (res instanceof Promise) {
+                      } else {
+                        setCommands(res ?? []);
+                      }
+                    }
+                  : debounce(async (e) => {
+                      setSelected(0);
+                      const query = e.target.value;
+                      setQuery(query);
+
+                      const res = search(query);
+                      if (res instanceof Promise) {
```

Comment:
I don't think this is a good idea. Make it all `async` and get rid of this `if..else` conditions.